### PR TITLE
glance: add uid and gid check

### PR DESCRIPTION
**Pull request by matelakat**
Make the recipe fail if the node is not compatible with hardwired
uid/gid/username/groupname constraints.

### CODE REVIEW

**Comment by nicolasbock** [comment]
@matelakat, @vuntz: I tried it out and it works as advertised by @matelakat. However, when a user re-installs a node and that node has a mismatching uid/gid the node ends up in 'problem' state without indication on the crowbar UI what went wrong. The traceback with the error message goes to `/var/log/crowbar/crowbar_join/chef.log` on that node (which is picked up by supportconfig) and is not visible directly on the crowbar node.

**Comment by nicolasbock** [comment]
The CI failure is related to nova and not to this PR.

**Comment by nicolasbock** [comment]
Fixed hound complaints.

**Comment by nicolasbock** [comment]
@vuntz Could you have a look at the last commit? This commit would address your comment regarding what to do with previously installed nodes. For this to work it is necessary to stop all glance service before running `usermod`. So far I haven't quite figured out how to do that.

**Comment by dirkmueller** [comment]
[P2: bsc#1004514: L3: glance uid: changed after reinstallation of a cluster node](https://trello.com/c/45uLzu6M/172-p2-bsc-1004514-l3-glance-uid-changed-after-reinstallation-of-a-cluster-node)

**Comment by Itxaka** [comment]
Shouldnt we update this to do it in a more chef-i way by setting user/group resources like @nicolasbock did in the rabbitmq cookbook?

File: chef/cookbooks/glance/recipes/common.rb [hunk 1]
```diff
@@ -22,6 +22,55 @@
...
+      Chef::Log.info("error from ajdusting glance uid/gid")
+      cmd.stderr.split("\n").each do |l|
+        Chef::Log.info(l)
+      end
+    end
+    if not cmd.stdout.empty?
+      Chef::Log.info("output from ajdusting glance uid/gid")
+      cmd.stdout.split("\n").each do |l|
+        Chef::Log.info(l)
+      end
```
Comment (vuntz) [inline chef/cookbooks/glance/recipes/common.rb]: Do we really need to put the output in the chef logs?

**Comment by nicolasbock** [comment]
>Does this really do what is intended? I mean, `notifies` is not 
>supposed to work from here.

Good point. I moved the notifies statement outside the `block do` 
block.

>Do we really need to put the output in the chef logs?

I removed those logging statements.

File: chef/cookbooks/glance/recipes/common.rb [hunk 2]
```diff
@@ -22,6 +22,29 @@
   package_name "openstack-glance" if %w(rhel suse).include?(node[:platform_family])
 end
 
+ruby_block "Adjust glance uid/gid" do
```
Comment (matelakat) [inline chef/cookbooks/glance/recipes/common.rb]: Do we really want to change gids and uids? What's going to happen with files that were already created, having the old gid/uid?

**Comment by matelakat** [comment]
@nicolasbock It might be a good idea to re-open this PR with your name?

**Comment by nicolasbock** [comment]
@matelakat I agree, I'll do that.

**Comment by nicolasbock** [comment]
Closing in favor of #800.

**Pull request closed without merging**